lucetc/src/output.rs

```diff
@@ -43,7 +73,22 @@ impl ObjectFile {
         let trap_manifest = &product
             .trap_manifest
             .expect("trap manifest will be present");
-        write_trap_manifest(trap_manifest, &mut product.artifact)?;
+
+        // TODO: at this moment there is no way to get a full list of functions and sizes
+        // at this point in compilation.
+        //
+        // For now, we need the list of functions with traps, which we can get here, and
+        // reuse that when writing out the trap manifest. When a full function list is
+        // available, `write_function_manifest` should take the function manifest, rather
+        // than compute it
+        let function_manifest = write_function_manifest(trap_manifest, &mut product.artifact)?;
+        let internal_trap_manifest = write_trap_tables(trap_manifest, &mut product.artifact)?;
```

Comment:
`write_trap_tables` happens to not need `function_manifest` as a coincidence at this moment, but if `function_manifest` were a full list, or reordered, it would be necessary to look up functions there.

I'll actually change `write_trap_manifest` to assume `function_manifest` might be any ol list, just so this isn't an issue in the future

Reply:
I think traps can just be a member of each function manifest. That isn't the design we wrote down originally, but I don't see any reason for them to be different, do you?

Reply:
Only if some data associated with a function might be sparse enough to justify not wanting a bunch of null pointers. Since most functions have some possibility of trapping, I agree